### PR TITLE
Migrate "with_winit" examples to winit's new "app" API

### PR DESCRIPTION
**Pull request by yutannihilation**
Second round of #626

Currently, this crashes with the following on my local, so I think I'm doing something wrong.

```
     Running `target\debug\with_winit_bin.exe`
[2024-07-15T15:33:30.317Z ERROR wgpu_hal::vulkan::instance] GENERAL [Loader Message (0x0)]
        loader_scanned_icd_add: Driver C:\Windows\System32\DriverStore\FileRepository\u0379858.inf_amd64_1719b9caae095668\B377432\.\amdvlk64.dll says it supports interface version 6 but still exports core entrypoints (Policy #LDP_DRIVER_6)
[2024-07-15T15:33:30.318Z ERROR wgpu_hal::vulkan::instance]     objects: (type: INSTANCE, hndl: 0x1ca928fbea0, name: ?)
Initialising in parallel using 14 threads
Parsed svg Ghostscript_Tiger in 8.3298ms
Encoded svg Ghostscript_Tiger in 893.9µs
thread 'main' panicked at examples\with_winit\src\lib.rs:511:22:
failed to get surface texture: Timeout
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: process didn't exit successfully: `target\debug\with_winit_bin.exe` (exit code: 101)

```

### CODE REVIEW

File: examples/with_winit/src/lib.rs [hunk 1]
```diff
@@ -88,30 +92,540 @@ struct RenderState<'s> {
...
+                    height,
+                    antialiasing_method,
+                };
+                self.scene.reset();
+                if let Some(resolution) = scene_params.resolution {
+                    // Automatically scale the rendering to fill as much of the window as possible
+                    // TODO: Apply svg view_box, somehow
+                    let factor = Vec2::new(width as f64, height as f64);
+                    let scale_factor = (factor.x / resolution.x).min(factor.y / resolution.y);
+                    self.transform *= Affine::scale(scale_factor);
```
Comment (DJMcNab) [inline examples/with_winit/src/lib.rs:467]: Why are you modifying `self.transform` here?

Changing this to match the prior logic resolves the crashing issue for me.

**Review by DJMcNab — COMMENTED** [review]
Other than the logic you've changed, this works fine for me.

File: examples/with_winit/src/lib.rs (hunk 1 above)
Comment (yutannihilation) [reply examples/with_winit/src/lib.rs:467]: Thanks for catching this! Fixing this mistake works for me, too.

**Comment by yutannihilation** [comment]
Thanks for reviewing!

**Pull request merged**